src/large_pages/node_large_page.cc

```diff
@@ -141,9 +147,66 @@ static struct text_region FindNodeTextRegion() {
   }
 
   ifs.close();
+#elif defined(__FreeBSD__)
+  struct text_region nregion;
+  nregion.found_text_region = false;
+
+  std::string exename;
+  {
+    char selfexe[PATH_MAX];
+    size_t count;
+    int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+    const size_t miblen = sizeof(mib) / sizeof(mib[0]);
+    if (sysctl(mib, miblen, selfexe, &count, nullptr, 0) == -1) {
+      return nregion;
+    }
+    exename = std::string(selfexe, count);
+  }
+
+  size_t iprocsz;
+  int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_VMMAP, getpid()};
+  const size_t miblen = sizeof(mib) / sizeof(mib[0]);
+  if (sysctl(mib, miblen, nullptr, &iprocsz, nullptr, 0) == -1) {
+    return nregion;
+  }
+
+  auto alg = std::vector<char>(iprocsz);
+
+  iprocsz = iprocsz * 4 / 3;
```

Comment:
Why is this necessary?

Reply:
it is for struct kinfo_vmentry.

Reply:
I'm not sure I follow. `alg` is `iprocsz` bytes big and next you tell the kernel it's `iprocsz * 1.25` bytes big (if I read the code correctly)

Reply:
Yes libpvocstat has similar workflow and also based on the code I wrote for radare a while ago
https://github.com/radare/radare2/blob/master/libr/debug/p/debug_native.c#L994